src/TypescriptParser.ts

```diff
@@ -71,28 +79,55 @@ export class TypescriptParser {
 
     /**
      * Parses multiple files into parsed files.
-     * 
+     *
      * @param {string[]} filePathes
      * @param {string} rootPath
      * @returns {Promise<File[]>}
-     * 
+     *
      * @memberof TsResourceParser
      */
-    public async parseFiles(filePathes: string[], rootPath: string): Promise<File[]> {
+    public async parseFiles(
+        filePathes: string[],
+        rootPath: string): Promise<File[]> {
         return filePathes
-            .map(o => createSourceFile(o, readFileSync(o).toString(), ScriptTarget.ES2015, true))
+            .map((o) => {
+                let scriptKind: ScriptKind = ScriptKind.TS;
+                const fileType: string = o.split('.').length ? <string>o.split('.').pop() : 'unknown';
```

Comment:
the package `path` does contain a method `parse`. I'd use the path.parse method to determine the file extension.
Which would result in:
```typescript
import { parse } from 'path';
const parsed = parse(o);
let kind = ScriptKind.Unknown;
switch(parsed.extension) {
    // the extensions
   default:
   break;
}
```

Reply:
or even `path.extname`